lumibot/tools/polygon_helper.py

```diff
@@ -480,3 +463,72 @@ def update_polygon_data(df_all, result):
             df_all = df_all[~df_all.index.duplicated(keep="first")]  # Remove any duplicate rows
 
     return df_all
+
+class PolygonClient(RESTClient):
+    ''' Rate Limited RESTClient with factory method '''
+
+    WAIT_SECONDS = 12
+
+    @classmethod
+    def create(cls, *args, **kwargs) -> RESTClient:
+        """
+        Factory method to create a RESTClient or PolygonClient instance.
+
+        The method uses environment variables to determine default values for the API key 
+        and subscription type. If the `api_key` is not provided in `kwargs`, it defaults 
+        to the value of the `POLYGON_API_KEY` environment variable. Similarly, the 
+        `paid` parameter defaults to the value of the `POLYGON_IS_PAID_SUBSCRIPTION` 
+        environment variable, which can be set to "true", "1", "t", "y", or "yes" to 
+        indicate a paid subscription. If the environment variable is not set, it defaults to False.
+
+        Keyword Arguments:
+        api_key : str, optional
+            The API key to authenticate with the service. Defaults to the value of the 
+            `POLYGON_API_KEY` environment variable if not provided.
+        paid : bool, optional
+            If False, a PolygonClient (rate limited) is created.
+            If True, a standard RESTClient is created. Default is False if the 
+            `POLYGON_IS_PAID_SUBSCRIPTION` environment variable is not set.
+
+        Returns:
+        RESTClient
+            An instance of RESTClient or PolygonClient.
+
+        Examples:
+        ---------
+        Using default environment variables:
+        
+        >>> client = PolygonClient.create()
+        
+        Providing an API key explicitly:
+        
+        >>> client = PolygonClient.create(api_key='your_api_key_here')
+        
+        Indicating a paid subscription explicitly:
+        
+        >>> client = PolygonClient.create(paid=True)
+        
+        Providing both an API key and indicating a paid subscription:
+        
+        >>> client = PolygonClient.create(api_key='your_api_key_here', paid=True)
+        """
+        POLYGON_API_KEY = os.environ.get("POLYGON_API_KEY")
+        POLYGON_IS_PAID_SUBSCRIPTION = os.getenv("POLYGON_IS_PAID_SUBSCRIPTION", "false").lower() in {'true', '1', 't', 'y', 'yes'}
+
+        kwargs['api_key'] = kwargs.get('api_key', POLYGON_API_KEY)
```

Comment:
I find it weird to be accessing environment variables inside a library function like this.  I also still don't really get why you're using args and kwargs dicts here instead of named args but I'll assume your reason is good.

How about as a compromise on the env access: 
Instead of use the default value in the the dict.get to check for the arg key in kwargs and if it is missing then get from os.environ.  If the arg is present then things work normally and the environ is not access.

Reply:
Yeah, might be a little weird but that would allow removing paid and api_key parameters from a lot of methods, and also redundant env calls if needed. And the args+kwargs is for not depending on API changes in _get method in polygon-api-client

Ok, so you want change this:

POLYGON_API_KEY = os.environ.get("POLYGON_API_KEY")
POLYGON_IS_PAID_SUBSCRIPTION = os.getenv("POLYGON_IS_PAID_SUBSCRIPTION", "false").lower() in {'true', '1', 't', 'y', 'yes'}
kwargs['api_key'] = kwargs.get('api_key', POLYGON_API_KEY)

to this? 

kwargs['api_key'] = kwargs.get('api_key', os.environ.get("POLYGON_API_KEY")
paid = kwargs.pop('paid', os.getenv("POLYGON_IS_PAID_SUBSCRIPTION", "false").lower() in {'true', '1', 't', 'y', 'yes'})

Seems reasonable, I'm doing it!

Reply:
ok, I get it! Then like this?
```
if 'api_key' not in kwargs:
        kwargs['api_key'] = os.environ.get("POLYGON_API_KEY")
        
if 'paid' not in kwargs:
        paid = os.getenv("POLYGON_IS_PAID_SUBSCRIPTION", "false").lower() in {'true', '1', 't', 'y', 'yes'}
else:
        paid = kwargs.pop('paid')
```

Reply:
Yup